src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SdkSupportedTargetPlatformVersionProvider.cs

```diff
@@ -0,0 +1,125 @@
+            // for the project.
+            EnsureInitialized();
+        }
+
+        protected override IDisposable? LinkExternalInput(ITargetBlock<EnumCollectionProjectValue> targetBlock)
+        {
+            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _subscriptionService.ProjectRuleSource;
+
+            // Transform the changes from design-time build -> supported target OS
+            DisposableValue<ISourceBlock<EnumCollectionProjectValue>> transformBlock = source.SourceBlock.TransformWithNoDelta(
+                update => update.Derive(Transform),
+                suppressVersionOnlyUpdates: false,
+                ruleNames: SdkSupportedTargetPlatformVersion.SchemaName);
+
+            // Set the link up so that we publish changes to target block
+            transformBlock.Value.LinkTo(targetBlock, DataflowOption.PropagateCompletion);
+
+            // Join the source blocks, so if they need to switch to UI thread to complete 
+            // and someone is blocked on us on the same thread, the call proceeds
+            JoinUpstreamDataSources(source);
+
+            return transformBlock;
+        }
+
+        private static EnumCollection Transform(IProjectSubscriptionUpdate input)
+        {
+            IProjectRuleSnapshot snapshot = input.CurrentState[SdkSupportedTargetPlatformVersion.SchemaName];
+
+            return snapshot.Items.Select(ToEnumValue)
+                                    .ToList();
+        }
+
+        private static IEnumValue ToEnumValue(KeyValuePair<string, IImmutableDictionary<string, string>> item)
+        {
+            return new PageEnumValue(new EnumValue()
+            {
+                // Example: <SdkSupportedTargetPlatformVersion Include="7.0"/>
+                //          <SdkSupportedTargetPlatformVersion Include="8.0"/>
+
+                Name = item.Key
+            });
+        }
+
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(this);
+        }
+
+        public async Task<EnumCollection> GetListedValuesAsync()
+        {
+            if (!IsReadyToBuild())
+                throw new InvalidOperationException("This configuration is not set to build");
+
+            // NOTE: This has a race, if called off the UI thread, the configuration could become 
+            // inactive underneath us and hence not ready for build, causing below to block forever.
+
+            using (JoinableCollection.Join())
+            {
+                EnumCollectionProjectValue snapshot = await SourceBlock.ReceiveAsync();
+
+                return snapshot.Value;
+            }
+        }
+
+        private bool IsReadyToBuild()
+        {
+            IConfiguredProjectReadyToBuild? readyToBuild = ReadyToBuild.FirstOrDefault()?.Value;
+
+            return readyToBuild?.IsValidToBuild == true;
+        }
```

Comment:
Could be written as:
```suggestion
        private bool IsReadyToBuild() => ReadyToBuild.FirstOrDefault()?.IsValidToBuild ?? false;
```

Reply:
This code ended up being deleted, but thanks.